src/mpid/ch4/netmod/ofi/ofi_spawn.c

```diff
@@ -318,6 +318,10 @@ int MPIDI_OFI_insert_upid(MPIR_Lpid lpid, const char *upid, int upid_len)
         MPIR_Assert(MPIDI_OFI_AV_ADDR_ROOT(av) != FI_ADDR_NOTAVAIL);
     }
 
+    if (MPIDI_OFI_AV_ADDR_ROOT(av) == 0) {
+        MPIDI_OFI_global.lpid0 = lpid;
+    }
```

Comment:
do we always want to set this even when we aren't actually inserting into the av?

Reply:
I believe you are right. Added a commit address this and also tried to make it more readable with macro and comments.

Reply:
Actually `lpid0` should be the formal (non-dynamic) lpid that has an `addr` equal to `0`. So we also need update it if it is not inserted here but being first time copied over from a dynamic entry.